File: core/src/main/java/io/github/fourlastor/game/demo/round/AbilityState.java
package io.github.fourlastor.game.demo.round;

import com.badlogic.gdx.ai.fsm.State;
import com.badlogic.gdx.ai.msg.Telegram;
import io.github.fourlastor.game.demo.state.GameState;

public abstract class AbilityState implements State<GameState> {

    @Override
    public abstract void enter(GameState state);

    @Override
    public void exit(GameState state) {}

    @Override
    public void update(GameState state) {}

    @Override
    public boolean onMessage(GameState state, Telegram telegram) {
        return false;
    }
}

File: core/src/main/java/io/github/fourlastor/game/demo/round/AbilityState.kt
package io.github.fourlastor.game.demo.round

import com.badlogic.gdx.ai.fsm.State
import com.badlogic.gdx.ai.msg.Telegram
import io.github.fourlastor.game.demo.state.GameState

abstract class AbilityState : State<GameState> {
  abstract override fun enter(state: GameState)
  override fun exit(state: GameState) {}
  override fun update(state: GameState) {}
  override fun onMessage(state: GameState, telegram: Telegram): Boolean = false
}
